### PR TITLE
Use almalinux as the Docker base image

### LINKED ISSUE #76681: Switch to future-proof linux distribution for Docker images
Centos 8 support updates will cease at the end of this year so we need to find a new Linux distro to base our docker images. There are obviously no shortage of options here but ideally:

1. Something with a good LTS scheme
2. Something well established (probably avoid the new up-and-coming CentOS replacements)
3. Ideally something RPM-based to align with UBI (not a hard requirement)

### PR DESCRIPTION
**Pull request by pugnascotia**
Closes #76681. Our approach to using `scratch` for building Docker images has caused problems at Docker Hub. Fix this situation by removing the whole process of using scratch and instead bases the default distribution on `almalinux:8.4-minimal`. Alma Linux is binary-compatible with RHEL, and therefore very similar to UBI.

cc @bytebilly @mieciu

### CODE REVIEW

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-windows - looks like transient failures

File: distribution/docker/src/docker/Dockerfile [hunk 1]
```diff
@@ -19,7 +19,7 @@
   add as many newlines here as necessary to improve legibility.
 */ %>
 
-<% if (docker_base == "ubi") { %>
+<% if (docker_base == 'default' || docker_base == "ubi") { %>
```
Comment (mark-vieira) [inline distribution/docker/src/docker/Dockerfile:22]: I think one of the issues the Dockerhub folks had was with the fact we're using a multi-stage build. Can we eliminate this given all we're doing now in this stage is unpacking ES and adding tini?

File: docs/reference/setup/install/docker.asciidoc [hunk 2]
```diff
@@ -2,8 +2,8 @@
 === Install {es} with Docker
 
 {es} is also available as Docker images. Starting with version 8.0.0, these
-are based upon a tiny core of essential files. Prior versions used
-https://hub.docker.com/_/centos/[centos:8] as the base image.
+are based upon
```
Comment (mark-vieira) [inline docs/reference/setup/install/docker.asciidoc:5]: I wondering if this should even be in user facing documentation at all.

**Review by mark-vieira — APPROVED** [review]
Couple comments but otherwise LGTM. Thanks for the quick turnaround on this, Rory.

File: distribution/docker/src/docker/Dockerfile (hunk 1 above)
Comment (pugnascotia) [reply distribution/docker/src/docker/Dockerfile:22]: I'll look into this in a separate PR.

**Comment by pugnascotia** [comment]
Closes #76681. Our approach to using `scratch` for building Docker
images has caused problems at Docker Hub. Fix this situation by
removing the whole process of using scratch and instead bases the
default distribution on `almalinux:8.4-minimal`. Alma Linux is
binary-compatible with RHEL, and therefore very similar to UBI.

**Comment by pugnascotia** [comment]
Backported to `7.16 in 4f17b3db514 and d6472380b50.

**Comment by pugnascotia** [comment]
I had to make another fix to the Dockerfile in 46dc92d.

**Pull request merged**